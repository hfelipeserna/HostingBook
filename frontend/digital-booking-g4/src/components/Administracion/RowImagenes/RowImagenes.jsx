import { useState } from "react";
import DropInput from "../DropInput/DropInput";
import EstandarInput from "../EstandarInput/EstandarInput";
import styles from "./RowImagenes.module.css";

export default function RowImagenes({
  imagenesDetails,
  handleAdd,
  handleDelete,
  agregarImagen,
}) {
 // TODO: Crear componente imagen y mover esto para allÃ¡
  const [imagenDetails, setImagenDetails] = useState({
    url: "",
    descripcion: "",
  });

  const setUrl = (url) =>
    setImagenDetails({
      url: url,
      descripcion: imagenDetails.descripcion,
    });

  const setDescripcion = (descripcion) =>
    setImagenDetails({
      url: imagenDetails.url,
      descripcion: descripcion,
    });

  return (
    <>
      {imagenesDetails.map((val, idx) => (
        <Imagen
          value={val}
          index={idx}
          agregarImagen={agregarImagen}
          handleAdd={handleAdd}
          handleDelete={handleDelete}
        />
      ))}
      {/* {imagenesDetails.map((val, idx) => {

        return (
          <div className={styles.fila}>
            <EstandarInput
              setOnChangeItem={setUrl}
              name="url"
              placeholder="Insertar https://"
              id={url}
              dataId={idx}
            />
            <DropInput
              setOnChangeItem={setDescripcion}
              name="descripcion"
              placeholder="DescripciÃ³n"
              id={descripcion}
              dataId={idx}
            />
            <div className={styles.containerButton}>
              {idx === 0 ? (
                <div className={styles.containerBotones}>
                  <button
                    className={styles.add}
                    onClick={() => {
                      agregarImagen(imagenDetails);
                    }}
                  >
                    <i className="fas fa-check"></i>
                  </button>
                  <button className={styles.add} onClick={handleAdd}>
                    <i className="fas fa-plus"></i>
                  </button>
                </div>
              ) : (
                <div className={styles.containerBotones}>
                  <button
                    className={styles.add}
                    onClick={() => {
                      agregarImagen(imagenDetails);
                    }}
                  >
                    <i className="fas fa-check"></i>
                  </button>
                  <button
                    className={styles.delete}
                    onClick={() => handleDelete(val)}
                  >
                    <i className="fas fa-times"></i>
                  </button>
                </div>
              )}
            </div>
          </div>
        );
      })} */}
    </>
  );
}
