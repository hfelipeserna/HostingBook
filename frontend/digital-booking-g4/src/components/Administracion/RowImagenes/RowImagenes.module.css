.fila{
    display: flex;
    flex-direction: row;
    align-items: flex-end;
    gap: 15px;
    background-color: rgba(190, 190, 190, 0.1);
    border-radius: 5px;
    padding: 0 15px 20px 15px;
    margin-top: 20px;


}

button{
    outline: none;
}

.add{
    font-size: 2rem;
    font-weight: 700;
    color: var(--color-5);
    cursor: pointer;
    background-color: var(--color-principal);
    border: 3px solid var(--color-principal);
    border-radius: 5px;
    filter: drop-shadow(0px 2px 4px rgba(0, 0, 0, 0.12));
    padding: 0 0.5vw;
    margin-top: 25px;
    transition: all ease-in-out 250ms;

}

.add:hover{
    background-color: var(--color-principal-opaco);
    border: 3px solid var(--color-principal-opaco);
}

.delete{
    font-size: 2rem;
    font-weight: 700;
    color: var(--color-5);
    cursor: pointer;
    background-color: var(--color-3);
    border: 3px solid var(--color-3);
    border-radius: 5px;
    filter: drop-shadow(0px 2px 4px rgba(0, 0, 0, 0.12));
    padding: 0 0.75vw;
    margin-top: 25px;
    transition: all ease-in-out 250ms;

}

.delete:hover{
    background-color: var(--color-2);
    border: 3px solid var(--color-2);
}

.containerBotones {
    display: flex;
    gap: 6px;
}

@media (max-width: 480px) {
    .fila{
        display: flex;
        flex-direction: column;
        gap: 0px;
    }
    .containerButton{
        width: 100%;
    }
    .add, .delete{
        width: 100%;
        margin-top: 9px;
    }
}