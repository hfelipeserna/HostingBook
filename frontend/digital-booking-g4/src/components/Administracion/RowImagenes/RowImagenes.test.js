
import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/react"
import RowImagenes from "./RowImagenes";






describe("Row Imagenes test", function () {

    const imagenesDetails = ["asdas", "asdasdas", "asdasd", "asdasd"]
    const handleAdd = () => {

    }
    const handleDelete = () => {

    }
    const agregarImagen = () => {

    }


    test("Renderiza Componente Por default", () => {
        const renderizacionConVariables = render(<RowImagenes agregarImagen={agregarImagen} handleAdd={handleAdd} handleDelete={handleDelete} imagenesDetails={imagenesDetails}></RowImagenes>)

        expect(renderizacionConVariables).toMatchSnapshot()

    });



});