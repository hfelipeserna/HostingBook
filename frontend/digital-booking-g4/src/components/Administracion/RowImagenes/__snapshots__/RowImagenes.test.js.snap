// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row Imagenes test Renderiza Componente Por default 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="fila"
        >
          <div
            class="containerEstandarInput"
          >
            <label
              class="labelEstandarInput"
            />
            <div
              class="sombraConAutocompletado"
            >
              <input
                class="estandarInput"
                name="url"
                placeholder="Insertar https://"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="containerDropInput"
          >
            <label
              class="labelDropInput"
            />
            <div
              class="cityContainer divDrawer"
            >
              
              <input
                autocomplete="off"
                class="input"
                placeholder="Descripción"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="containerButton"
          >
            <div
              class="containerBotones"
            >
              <button
                class="add"
                disabled=""
              >
                <i
                  class="fas fa-check"
                />
              </button>
              <button
                class="add"
              >
                <i
                  class="fas fa-plus"
                />
              </button>
            </div>
          </div>
        </div>
        
      </div>
      <div
        class="container"
      >
        <div
          class="fila"
        >
          <div
            class="containerEstandarInput"
          >
            <label
              class="labelEstandarInput"
            />
            <div
              class="sombraConAutocompletado"
            >
              <input
                class="estandarInput"
                name="url"
                placeholder="Insertar https://"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="containerDropInput"
          >
            <label
              class="labelDropInput"
            />
            <div
              class="cityContainer divDrawer"
            >
              
              <input
                autocomplete="off"
                class="input"
                placeholder="Descripción"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="containerButton"
          >
            <div
              class="containerBotones"
            >
              <button
                class="add"
                disabled=""
              >
                <i
                  class="fas fa-check"
                />
              </button>
              <button
                class="delete"
              >
                <i
                  class="fas fa-times"
                />
              </button>
            </div>
          </div>
        </div>
        
      </div>
      <div
        class="container"
      >
        <div
          class="fila"
        >
          <div
            class="containerEstandarInput"
          >
            <label
              class="labelEstandarInput"
            />
            <div
              class="sombraConAutocompletado"
            >
              <input
                class="estandarInput"
                name="url"
                placeholder="Insertar https://"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="containerDropInput"
          >
            <label
              class="labelDropInput"
            />
            <div
              class="cityContainer divDrawer"
            >
              
              <input
                autocomplete="off"
                class="input"
                placeholder="Descripción"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="containerButton"
          >
            <div
              class="containerBotones"
            >
              <button
                class="add"
                disabled=""
              >
                <i
                  class="fas fa-check"
                />
              </button>
              <button
                class="delete"
              >
                <i
                  class="fas fa-times"
                />
              </button>
            </div>
          </div>
        </div>
        
      </div>
      <div
        class="container"
      >
        <div
          class="fila"
        >
          <div
            class="containerEstandarInput"
          >
            <label
              class="labelEstandarInput"
            />
            <div
              class="sombraConAutocompletado"
            >
              <input
                class="estandarInput"
                name="url"
                placeholder="Insertar https://"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="containerDropInput"
          >
            <label
              class="labelDropInput"
            />
            <div
              class="cityContainer divDrawer"
            >
              
              <input
                autocomplete="off"
                class="input"
                placeholder="Descripción"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="containerButton"
          >
            <div
              class="containerBotones"
            >
              <button
                class="add"
                disabled=""
              >
                <i
                  class="fas fa-check"
                />
              </button>
              <button
                class="delete"
              >
                <i
                  class="fas fa-times"
                />
              </button>
            </div>
          </div>
        </div>
        
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="fila"
      >
        <div
          class="containerEstandarInput"
        >
          <label
            class="labelEstandarInput"
          />
          <div
            class="sombraConAutocompletado"
          >
            <input
              class="estandarInput"
              name="url"
              placeholder="Insertar https://"
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="containerDropInput"
        >
          <label
            class="labelDropInput"
          />
          <div
            class="cityContainer divDrawer"
          >
            
            <input
              autocomplete="off"
              class="input"
              placeholder="Descripción"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="containerButton"
        >
          <div
            class="containerBotones"
          >
            <button
              class="add"
              disabled=""
            >
              <i
                class="fas fa-check"
              />
            </button>
            <button
              class="add"
            >
              <i
                class="fas fa-plus"
              />
            </button>
          </div>
        </div>
      </div>
      
    </div>
    <div
      class="container"
    >
      <div
        class="fila"
      >
        <div
          class="containerEstandarInput"
        >
          <label
            class="labelEstandarInput"
          />
          <div
            class="sombraConAutocompletado"
          >
            <input
              class="estandarInput"
              name="url"
              placeholder="Insertar https://"
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="containerDropInput"
        >
          <label
            class="labelDropInput"
          />
          <div
            class="cityContainer divDrawer"
          >
            
            <input
              autocomplete="off"
              class="input"
              placeholder="Descripción"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="containerButton"
        >
          <div
            class="containerBotones"
          >
            <button
              class="add"
              disabled=""
            >
              <i
                class="fas fa-check"
              />
            </button>
            <button
              class="delete"
            >
              <i
                class="fas fa-times"
              />
            </button>
          </div>
        </div>
      </div>
      
    </div>
    <div
      class="container"
    >
      <div
        class="fila"
      >
        <div
          class="containerEstandarInput"
        >
          <label
            class="labelEstandarInput"
          />
          <div
            class="sombraConAutocompletado"
          >
            <input
              class="estandarInput"
              name="url"
              placeholder="Insertar https://"
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="containerDropInput"
        >
          <label
            class="labelDropInput"
          />
          <div
            class="cityContainer divDrawer"
          >
            
            <input
              autocomplete="off"
              class="input"
              placeholder="Descripción"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="containerButton"
        >
          <div
            class="containerBotones"
          >
            <button
              class="add"
              disabled=""
            >
              <i
                class="fas fa-check"
              />
            </button>
            <button
              class="delete"
            >
              <i
                class="fas fa-times"
              />
            </button>
          </div>
        </div>
      </div>
      
    </div>
    <div
      class="container"
    >
      <div
        class="fila"
      >
        <div
          class="containerEstandarInput"
        >
          <label
            class="labelEstandarInput"
          />
          <div
            class="sombraConAutocompletado"
          >
            <input
              class="estandarInput"
              name="url"
              placeholder="Insertar https://"
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="containerDropInput"
        >
          <label
            class="labelDropInput"
          />
          <div
            class="cityContainer divDrawer"
          >
            
            <input
              autocomplete="off"
              class="input"
              placeholder="Descripción"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="containerButton"
        >
          <div
            class="containerBotones"
          >
            <button
              class="add"
              disabled=""
            >
              <i
                class="fas fa-check"
              />
            </button>
            <button
              class="delete"
            >
              <i
                class="fas fa-times"
              />
            </button>
          </div>
        </div>
      </div>
      
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
